Kernel/Graphics/VirtIOGPU/GPU3DDevice.h

```diff
@@ -101,24 +103,35 @@ class VirtIOGPU3DDevice : public CharacterDevice {
     VirtIOGPU3DDevice(VirtIOGraphicsAdapter const& graphics_adapter, NonnullOwnPtr<Memory::Region> transfer_buffer_region, Graphics::VirtIOGPU::ContextID kernel_context_id);
 
     class PerContextState final : public AtomicRefCounted<PerContextState> {
+        friend class VirtIOGPU3DDevice;
+
     public:
-        static ErrorOr<LockRefPtr<PerContextState>> try_create(Graphics::VirtIOGPU::ContextID context_id)
+        static ErrorOr<NonnullRefPtr<PerContextState>> try_create(OpenFileDescription& description, Graphics::VirtIOGPU::ContextID context_id)
         {
             auto region_result = TRY(MM.allocate_kernel_region(
                 NUM_TRANSFER_REGION_PAGES * PAGE_SIZE,
                 "VIRGL3D userspace upload buffer"sv,
                 Memory::Region::Access::ReadWrite,
                 AllocationStrategy::AllocateNow));
-            return TRY(adopt_nonnull_lock_ref_or_enomem(new (nothrow) PerContextState(context_id, move(region_result))));
+            return TRY(adopt_nonnull_ref_or_enomem(new (nothrow) PerContextState(description, context_id, move(region_result))));
         }
         Graphics::VirtIOGPU::ContextID context_id() { return m_context_id; }
         Memory::Region& transfer_buffer_region() { return *m_transfer_buffer_region; }
 
+        OpenFileDescription& description() { return m_attached_file_description; }
+
     private:
         PerContextState() = delete;
-        explicit PerContextState(Graphics::VirtIOGPU::ContextID context_id, OwnPtr<Memory::Region> transfer_buffer_region);
+        PerContextState(OpenFileDescription&, Graphics::VirtIOGPU::ContextID context_id, OwnPtr<Memory::Region> transfer_buffer_region);
         Graphics::VirtIOGPU::ContextID m_context_id;
         OwnPtr<Memory::Region> m_transfer_buffer_region;
+
+        // NOTE: We clean this whole object when the file description is gone, therefore we can hold
+        // a raw reference here instead of a strong reference (RefPtr) which will make it impossible
+        // to not leak a OpenFileDescription being attached to a context in this device.
```

Comment:
"which will make it impossible to not leak a OpenFileDescription"

So this makes it possible to leak a description?

Reply:
It should have been:
```
// ... instead of a strong reference (RefPtr which will make it impossible
// to not leak a OpenFileDescription being attached to a context in this device)
```
so I will fix this now.